gossip/infostore.go

```diff
@@ -237,11 +241,34 @@ func (is *infoStore) processCallbacks(key string, content roachpb.Value) {
 		}
 	}
 
-	// Run callbacks in a goroutine to avoid mutex reentry.
-	go func() {
+	// Add the matching callbacks to the callback work list.
+	f := func() {
 		for _, method := range matches {
 			method(key, content)
 		}
+	}
+	is.callbackMu.Lock()
+	is.callbackWork = append(is.callbackWork, f)
+	is.callbackMu.Unlock()
+
+	// Run callbacks in a goroutine to avoid mutex reentry. We also guarantee
+	// callbacks are run in order such that if a key is updated twice in
+	// succession, the second callback will never be run before the first.
+	go func() {
+		// Grab the callback semaphore to prevent another goroutine from invoking
```

Comment:
I see what you're doing here, but I'm unfamiliar with this usage of the word "semaphore" (as a distinction from `callbackMu`). Renaming one or both variables or providing more comments might be helpful.